jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/Slider.java

```diff
@@ -0,0 +1,152 @@
+
+        String newTrackBackground = currentTrackBackground.substring(0, backgroundStartPoint+3) + newTrackBackgroundValue + currentTrackBackground.substring(backgroundEndPoint);
+        String newTrackFill = currentTrackFill.substring(0, fillStartPoint+3) + newTrackFillValue + currentTrackFill.substring(fillEndPoint).replace("display: none;", "");
+        String newThumbContainer = currentThumbContainer.substring(0, thumbStartPoint) + "-" + newThumbValue + currentThumbContainer.substring(thumbEndPoint);
+
+        sliderTrackBackground.setAttribute("style", newTrackBackground);
+        sliderTrackFill.setAttribute("style", newTrackFill);
+        sliderThumbContainer.setAttribute("style", newThumbContainer);
+    }
+
+    /**
+     *
+     * @param value
+     */
+    @Override
+    public void setValue(String value) {
+        uiElement.setAttribute("aria-valuenow", value);
+
+        UIElement sliderTrackBackground = this.uiElement.find(".mat-slider-track-background");
+        UIElement sliderTrackFill = this.uiElement.find(".mat-slider-track-fill");
+        UIElement sliderThumbContainer = this.uiElement.find(".mat-slider-thumb-container");
+
+        // set value for slider attributes to reflect changes on UI
+        reflectSliderChangeOnUI(Double.parseDouble(value), sliderTrackBackground, sliderTrackFill, sliderThumbContainer);
+    }
+
+    /**
+     * drag & drop based on percentage length of slider
+     * @param percentage
+     */
+    public void slide(double percentage) {
+        int xOffset = (int) Math.round(percentage * getStepInPixels(this.uiElement));
+        this.uiElement.find(".mat-slider-thumb").dragAndDropTo(xOffset, 0);
+    }
+
+    private Double getStepInPixels(UIElement uiElement) {
+        return uiElement.getSize().width / 100.0;
+    }
+
+    /**
+     * check if slider is vertical
+     * @return
+     */
+    public boolean isVertical() {
+        return this.uiElement.attr("aria-orientation").equals("vertical");
+    }
+
+    /**
+     * check if slider is disabled
+     * @return
+     */
+    public boolean isDisabled() {
+        return this.uiElement.attr("aria-disabled").equals("true");
+    }
+
+    public boolean isEnabled() {
+        return !isDisabled();
+    }
+
+    /**
+     * check if slider is inverted
+     * @return
+     */
+    public boolean isInverted() {
+        return this.uiElement.hasClass("mat-slider-axis-inverted");
+    }
+
+    /**
+     * check if Thumb Label is displayed
+     * @return
+     */
+    public boolean isThumbLabelDisplayed() {
+        return this.uiElement.find(".mat-slider-thumb-label-text").isVisible();
+    }
+
+    /**
+     * move slider to right
+     * @return
+     */
+    public void moveRight() {
+        this.uiElement.sendKeys(Keys.ARROW_RIGHT);
+    }
+
+    /**
+     * move slider to left
+     * @return
+     */
+    public void moveLeft() {
+        this.uiElement.sendKeys(Keys.ARROW_LEFT);
```

Comment:
Replace `this.uiElement `with just `uiElement`  here and in other places in this class

Reply:
fixed